### PR TITLE
(CAT-1430) - Drop ruby 2.5 Support & Add Ruby 3.x Support

### PR DESCRIPTION
**Pull request by jordanbreen28**
## Summary
Drops support for ruby 2.5.

Adds support for ruby 3.x

Bumps the runtime dependency gems for puppet-lint and puppet-strings to latest versions.
puppet-lint development gem bumped to `~> 4.0`. 

See gemfile:
```
# There should be NO runtime gem dependencies here.  In production this code will be running using the Ruby
# runtime provided by Puppet.  That means no native extensions and NO BUNDLER.  All runtime dependences should
# be re-vendored and then the load path modified appropriately.
#
# This gemfile only exists to help when developing the language server and running tests
```
Temporarily vendor gem from https://github.com/jordanbreen28/puppetfile-resolver until https://github.com/glennsarti/puppetfile-resolver/pull/41 is merged/released.

## Additional Context
Add any additional context about the problem here. 
- [ ] Root cause and the steps to reproduce. (If applicable)
- [ ] Thought process behind the implementation.

## Related Issues (if any)
Mention any related issues or pull requests.

## Checklist
- [ ] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [ ] Manually verified.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -25,14 +25,19 @@ jobs:
       ruby_version: ${{ matrix.ruby_version }}
       runs_on: ${{ matrix.runs_on }}
       rake_task: 'gem_revendor test_languageserver test_languageserver_sidecar test_debugserver'
-
+
```
Comment (GSPatton) [inline .github/workflows/nightly.yml:28]: very small thing... but looks like unneeded whitespace here

File: Rakefile [hunk 2]
```diff
@@ -64,8 +64,8 @@ task :gem_revendor do
     },
     {
       :directory => 'puppetfile-resolver',
-      :github_repo => 'https://github.com/glennsarti/puppetfile-resolver.git',
-      :github_ref => '0.3.0',
+      :github_repo => 'https://github.com/jordanbreen28/puppetfile-resolver.git',
```
Comment (GSPatton) [inline Rakefile:67]: i the plan to use your fork going forward?

Comment (GSPatton) [reply Rakefile:67]: i may have missed this yesterday

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (jordanbreen28) [reply .github/workflows/nightly.yml:28]: thank u sir

**Review by pmcmaw — APPROVED** [review]

**Pull request merged**